server/cluster/cluster.go

```diff
@@ -665,6 +664,56 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 		time.Sleep(500 * time.Millisecond)
 	})
 
+	return c.saveRegion(isNew, saveKV, saveCache, needSync, region, origin)
+}
+
+// processRegionHeartbeatResponse updates the region information thought region split.
+// It will update peer that not exist in the cache, and add new peers that join in.
+func (c *RaftCluster) processRegionSpiltReport(regions []*metapb.Region) (err error) {
+	if err = checkSplitRegions(regions); err != nil {
+		return err
+	}
+	c.RLock()
+	coreCluster := c.core
+	c.RUnlock()
+	total := len(regions) - 1
+	for i := range regions {
+		// It should update rightmost because it will override by left.
+		index := total - i
+		v := regions[index]
+		if v.GetPeers() == nil {
+			err = errors.Wrap(err, fmt.Sprintf("region:%d peer is nil", v.GetId()))
+			continue
+		}
+		region := core.NewRegionInfo(v, v.Peers[0])
```

Comment:
Why `v.Peers[0]`?

Reply:
It needs to resign one leader to get router but the leader may not elect.

Reply:
But in PD's view, it is the leader.